src/common/mfu_io.c

```diff
@@ -57,7 +232,27 @@ int mfu_lchown(const char* path, uid_t owner, gid_t group)
     return rc;
 }
 
-/* calls chmod, and retries a few times if we get EIO or EINTR */
+int daos_chmod(const char *path, mode_t mode, mfu_file_t* mfu_file)
```

Comment:
This is where the heavy lifting will be in this approach.  We either need to:
1) Create DAOS versions of the libc/posix functions so that they behave like posix functions with similar semantics, return code values, and setting of errno.  Then the code that makes calls to these will function assuming libc/posix.  For example, on chmod(2) it should return 0 on success, or -1 on failure and set errno appropriately.
2) Or we should define a new mfu_file_chmod replacement that by design does not follow libc/posix but instead has its own behavior that we define and control.  In that case, we'll need to review all code making this call and update it appropriately.

I don't have a good feel off hand for which way makes the most sense just yet,  I think option 1 might be a good start, but we can chat about it more.

Reply:
@adammoody I'm thinking (1) is the better option, and we can discuss more of this tomorrow.

Reply:
@adammoody The DFS API for DAOS does follow the POSIX return codes here: https://github.com/daos-stack/daos/blob/master/src/client/dfs/dfs.c, so the wrappers would be returning the correct codes, matching the current mfu_ calls.

Reply:
Excellent!  That will save a bunch of work then.